### DISCUSSION TITLE
How to build an older version of frida-server for iOS?

### DISCUSSION THREAD

**Discussion by hyunjong-prestolabs** [frida/frida#2736] [General]
Hi,
Currently, the frida repository doesn't provide any older `frida-server` binaries for IOS, so I want to build my own.

I cloned the frida repository locally and ran the make command, but the build failed with the following error, what should I do?
```
make[1]: *** No rule to make target `.git/refs/heads/master', needed by `build/frida-version.h'.  Stop.
make: *** [core-ios] Error 2
```

The sequence of operations is as follows
1. git clone <frida_repo>
2. git checkout <old_version>
3. make core-ios

Importantly, if I executed the make command without checking out to the old version after git clone, it worked fine.

**Comment by jpstotz** [comment]
The main frida repo https://github.com/frida/frida uses submodules, therefore if you just clone it you will miss a lot. To clone it with submodules use this command:

    git clone --recurse-submodules https://github.com/frida/frida

**Comment by ChiChou** [comment]
The best way is not to build.

Find a historical version, for example:
https://github.com/frida/frida/releases/tag/16.7.4

Search file name that looks like this, for example 64bit system:
[frida_16.7.4_iphoneos-arm64.deb](https://github.com/frida/frida/releases/download/16.7.4/frida_16.7.4_iphoneos-arm64.deb)

On macOS, unpack it:

```shell
$ tar xvf frida_16.7.4_iphoneos-arm64.deb
x debian-binary
x control.tar.xz
x data.tar.xz
```

Unpack `data.tar.xz`: `tar xvf data.tar.xz` and you get the binary

`./var/jb/usr/lib/frida/frida-agent.dylib`
`./var/jb/usr/sbin/frida-server`
